x-pack/plugins/security_solution/public/resolver/store/data/selectors.ts

```diff
@@ -167,6 +168,105 @@ export function hasMoreAncestors(state: DataState): boolean {
   return tree ? resolverTreeModel.hasMoreAncestors(tree) : false;
 }
 
+/**
+ * A map of `entity_id`s to functions that provide information about
+ * related events by ECS `.category` Primarily to avoid having business logic
+ * in UI components.
+ */
+export const relatedEventInfoByEntityId = createSelector(
```

Comment:
will the UI always need up to date `relatedEventInfo` objects for all entityIDs?

What about this type instead: 

```
export const relatedEventInfoByEntityId: (state: DataState) => (entityID: string) => RelatedEventInfo | undefined = createSelector
```

The body of the selector could be taken directly from [line 187](https://github.com/elastic/kibana/pull/70097/files#diff-8c9e76a944adc739544d6fb27f098079R187)

With this design, the selector would lazily create these objects as needed and garbage collect them when possible. This design might be more future proof. I expect 7.10 release to support something like 2-10k nodes.

Also, based on my understanding of this code, if a single node gets new related events, this will be recalculated for all events.

Reply:
can i do this as a perf. improvement in a followup?